signer/src/transaction_signer.rs

```diff
@@ -425,6 +535,125 @@ where
         Ok(())
     }
 
+    fn create_new_state_machine(&mut self) -> Result<SignerStateMachine, error::Error> {
```

Comment:
Also, it's a shame that this returns a `Result`. It seems like they are all phantom errors. No need to change this now, if we have time we can try to "fix up" the return type later.

Reply:
Yeah I agree. Right now I'd rather propagate "phantom errors" than introduce a potential panic in the system but it would be wonderful to make the interface reflect that this is infallible.

Reply:
Also thanks for spotting. I'm surprised the compiler didn't warn about this.

Reply:
Updated ✅